addon/components/base/bs-carousel.js

```diff
@@ -0,0 +1,467 @@
+      }
+    },
+    toPrevSlide() {
+      if (this.canTurnToLeft()) {
+        this.send('toSlide', this.get('index') - 1);
+      }
+    }
+  },
+  didInsertElement() {
+    this._super(...arguments);
+    this.populateIndicators();
+    this.registerEvents();
+    if (this.get('slides').length !== 0) {
+      this.get('slides')[this.get('index')].set('active', true);
+    }
+    if (this.shouldRunAutomatically() && this.get('autoPlay')) {
+      this.waitIntervalToInitCycle();
+    }
+  },
+  init() {
+    this._super(...arguments);
+    this.set('slides', A([]));
+  },
+
+  /**
+   * Transfers 'active' class name of current index to the next follwing index.
+   * 
+   * @method adjustIndicators
+   * @private
+   */
+  adjustIndicators() {
+    let indicators = this.get('indicators');
+    let currIndicator = indicators[this.get('index')];
+    let followingIndicator = indicators[this.get('followingSlideIndex')];
+    set(currIndicator, 'active', false);
+    set(followingIndicator, 'active', true);
+  },
+  /**
+   * Indicates what class names should be applicable to the current transition slides. 
+   * 
+   * @method assignClassNameControls
+   * @private
+   */
+  assignClassNameControls(toIndex) {
+    if (toIndex < this.get('index')) {
+      this.set('directionalClassName', 'right');
+      this.set('orderClassName', 'prev');
+    } else {
+      this.set('directionalClassName', 'left');
+      this.set('orderClassName', 'next');
+    }
+  },
+  /**
+   * @method canTurnToLeft
+   * @private
+   */
+  canTurnToLeft() {
+    let canNotTurn = this.get('continuouslyCycle') === false && this.get('index') <= 0;
+    return !canNotTurn;
+  },
+  /**
+   * @method canTurnToRight
+   * @private
+   */
+  canTurnToRight() {
+    let canNotTurn = this.get('continuouslyCycle') === false
+      && this.get('index') >= this.get('slides').length - 1;
+    return !canNotTurn;
+  },
+  /**
+   * Cancels a scheduled cycle item.
+   * 
+   * @method cancelCycle
+   * @private
+   */
+  cancelCycle() {
+    cancel(this.get('cycle'));
+  },
+  /**
+   * Do a presentation and calls itself to perform a cycle.
+   * 
+   * @method doCycle
+   * @private
+   */
+  doCycle() {
+    if (this.get('ltr') && this.canTurnToRight()) {
+      this.transition(this.get('index') + 1);
+    } else if (this.get('ltr') === false && this.canTurnToLeft()) {
+      this.transition(this.get('index') - 1);
+    } else {
+      return null;
+    }
+    return later(this, function() {
+      this.set('cycle', this.doCycle());
+    }, this.get('transitionDuration') + this.get('interval'));
+  },
+  /**
+   * Applies class names to slides, sets the current index as the following index and
+   * indicates that the transition has finished.
+   * 
+   * @method didTransition
+   * @private
+   */
+  didTransition(currSlide, followingSlide) {
+    currSlide.set('active', false);
+    currSlide.set(this.get('directionalClassName'), false);
+    followingSlide.set('active', true);
```

Comment:
Well, kept the same strategy since the slides instances play nicely with the component-(child|parent) mixins and I added a real time iterative carousel that allows dynamic inserting and removing.
The code is not ready to merge yet but I think we are getting close.